packages/contracts/test/registry/ens-subdomain-registry.ts

```diff
@@ -0,0 +1,253 @@
+      await ens
+        .connect(signers[0])
+        .setApprovalForAll(subdomainRegistrar.address, true);
+
+      expect(
+        await subdomainRegistrar
+          .connect(signers[0])
+          .initialize(dao.address, ens.address, namehash('test'))
+      );
+    });
+
+    it('initializes if the subdomain registrar owns the node', async () => {
+      // Register the parent domain 'test' through signers[0] who owns the ENS root node ('') and make the subdomain registrar the owner
+      registerSubdomainHelper(
+        'test',
+        '',
+        signers[0],
+        subdomainRegistrar.address
+      );
+
+      expect(
+        await subdomainRegistrar
+          .connect(signers[0])
+          .initialize(dao.address, ens.address, namehash('test'))
+      );
+    });
+
+    it('does not initialize if the registrar is neither the domain node owner nor an approved operator of the domain node owner', async () => {
+      await expect(
+        subdomainRegistrar
+          .connect(signers[0])
+          .initialize(dao.address, ens.address, namehash('test'))
+      ).to.be.revertedWith(
+        customError(
+          'RegistrarUnauthorized',
+          ethers.constants.AddressZero,
+          subdomainRegistrar.address
+        )
+      );
+    });
+  });
+
+  describe('registerSubnode:', () => {
+    it('registers and reassigns a subdomain if the registrar is the parent domain owner', async () => {
+      // Register the parent domain 'test' through signers[0] who owns the ENS root node ('') and make the subdomain registrar the owner
+      registerSubdomainHelper(
+        'test',
+        '',
+        signers[0],
+        subdomainRegistrar.address
+      );
+
+      // Initialize the registrar with the 'test' domain
+      subdomainRegistrar.initialize(dao.address, ens.address, namehash('test'));
+
+      // Grant signers[1] the right to register new subdomains in the subdomain registrar contract
+      await dao.grant(
+        subdomainRegistrar.address,
+        await signers[1].getAddress(),
+        REGISTER_ENS_SUBDOMAIN_ROLE
+      );
+
+      // Register the subdomain 'my.test' as signers[1] and give ownership to an address (here a DAO)
+      let tx = await subdomainRegistrar
+        .connect(signers[1])
+        .registerSubnode(labelhash('my'), dao.address);
+      tx.wait();
+      expect(await ens.owner(namehash('my.test'))).to.equal(dao.address);
+
+      // Reassigns the subdomain to another address (here signers[1].address)
+      tx = await subdomainRegistrar
+        .connect(signers[1])
+        .registerSubnode(labelhash('my'), signers[1].address);
+      tx.wait();
+      expect(await ens.owner(namehash('my.test'))).to.equal(signers[1].address);
+    });
+
+    it('registers and reassign a subdomain if the registrar is approved by the domain owner', async () => {
+      // Register the parent domain 'test' through signers[0] who owns the ENS root node ('') and make signers[1] its owner
```

Comment:
If the address of a DAO is always going to stay the same I'm not sure why we would want to allow changing the record.

Reply:
And as above, IMO, the owner of all subnodes should be just the registrar contract itself. Not target DAO's.

Reply:
I changed this and adapted the test.